README.md

```diff
@@ -1,32 +1,254 @@
+
+You need to locate/decide on the following before you create the package metadata config file:
+
+1. The output file to contain the package metadata for your client generation. Hereafter, this will be
+   referenced as `${PKG_META_CONFIG}`.
+2. A short name, which typically does not include the major version. For example, Cloud Language v1's short name
+   is `language`. Hereafter, this will be referenced as `${SHORT_NAME}`.
+3. The major version. The first major version will typically be `v1`. Hereafter, this will be referenced as
+   `${MAJOR_VERSION}`.
+4. The package name, which typically includes the fully-qualified product name and major version.
+   For example, Cloud Language v1's package name is `google-cloud-language-v1`. Hereafter, this will be
+   referenced as `${PACKAGE_NAME}`.
+5. The proto path, which is the proto's package with dots converted to slashes. For example, Cloud Language v1's
+   proto path is `google/cloud/language/v1`. Hereafter, this will be referenced as `${PROTO_PATH}`.
+
+First, copy the contents of api_defaults.yaml and dependencies.yaml into a single file, hereafter
+referenced as `${PKG_META_CONFIG}`.
+
+```
+cp ${GOOGLEAPIS_DIR}/gapic/packaging/api_defaults.yaml ${PKG_META_CONFIG}
+cat < ${GOOGLEAPIS_DIR}/gapic/packaging/dependencies.yaml >> ${PKG_META_CONFIG}
+```
+
+Add the following lines to the end:
+
+```
+artifact_type: GAPIC
+proto_deps:
+- google-common-protos
+short_name: ${SHORT_NAME}
+major_version: ${MAJOR_VERSION}
+package_name:
+  default: ${PACKAGE_NAME}
+proto_path: ${PROTO_PATH}
+```
+
+### Run code generation
+
+You need to locate/decide on the following before you call code generation:
+
+1. The language-specific generator config for your desired output language. Given a language where
+   `${language}` is in `["csharp", "go", "java", "nodejs", "php", "py", "ruby"]`, the file is located at
+   `src/main/resources/com/google/api/codegen/${language}/${language}_gapic.yaml` (exception: the Python file is named
+   `python_gapic.yaml`). Hereafter, this will be referenced as `${LANGUAGE_CONFIG}`.
+2. The output directory for your generated client classes. Hereafter, this will be
+   referenced as `${GENERATED_CLIENT_DIR}`.
+
+```
+java -cp build/libs/gapic-generator-*-fatjar.jar com.google.api.codegen.CodeGeneratorTool \
+  --descriptor_set=${YOUR_DESCRIPTOR_FILE} --service_yaml=${YOUR_SERVICE_YAML} \
+  --gapic_yaml=${YOUR_CLIENT_CONFIG} --gapic_yaml=${LANGUAGE_CONFIG} \
+  --package_yaml=${PKG_META_CONFIG} --o=${GENERATED_CLIENT_DIR}
+```
+
+The generated client library code will appear in `${GENERATED_CLIENT_DIR}`.
+
+You can safely ignore the warning about control-presence.
+
+Special note for java: several files will be dumped into the parent directory of `${GENERATED_CLIENT_DIR}`:
+
+- gradle/
+- gradlew
+- gradlew.bat
+
+(There is an open issue to fix this: https://github.com/googleapis/toolkit/issues/1918 )
+
+### Perform fix-ups to get a working library
```

Comment:
Here and below, I suggest:

s/fix-ups/fixes/
or
s/fix-ups/touch-ups/

Reply:
I will use "fixes".